tests/qos/test_qos_dscp_mapping.py

```diff
@@ -0,0 +1,238 @@
+        outer_src_pkt_ip = dutConfig["testPorts"]["src_port_ip"]
+        inner_dst_port_id = dutConfig["testPorts"]["dst_port_id"]
+        inner_dst_pkt_ip = dutConfig["testPorts"]["dst_port_ip"]
+        # inner_src_port_id = dutConfig["testPorts"]["src_port_id"]
+        inner_src_pkt_ip = DUMMY_IP
+        ptf_src_mac = ptfadapter.dataplane.get_mac(0, outer_src_port_id)
+        ptf_dst_mac = ptfadapter.dataplane.get_mac(0, inner_dst_port_id)
+        outer_dscp = 4
+
+        pytest_assert(dut_qos_maps.get("dscp_to_tc_map") and dut_qos_maps.get("tc_to_queue_map"),
+                      "No QoS map found on DUT")
+
+        for inner_dscp in range(0, 64):
+            pkt, exp_pkt = create_ipip_packet(outer_src_mac=ptf_src_mac,
+                                              outer_dst_mac=router_mac,
+                                              outer_src_pkt_ip=outer_src_pkt_ip,
+                                              outer_dst_pkt_ip=outer_dst_pkt_ip,
+                                              outer_dscp=outer_dscp,
+                                              inner_src_mac=router_mac,
+                                              inner_dst_mac=ptf_dst_mac,
+                                              inner_src_pkt_ip=inner_src_pkt_ip,
+                                              inner_dst_pkt_ip=inner_dst_pkt_ip,
+                                              inner_dscp=inner_dscp)
+
+            queue_val = get_dscp_to_queue_value(inner_dscp, dut_qos_maps.get("dscp_to_tc_map"),
+                                                dut_qos_maps.get("tc_to_queue_map"))
+            dut_egress_port = get_dut_pair_port_from_ptf_port(duthost, tbinfo, inner_dst_port_id)
+            pytest_assert(dut_egress_port, "No egress port on DUT found for ptf port {}".format(inner_dst_port_id))
+
+            # Clear queue counters
+            duthost.command("sonic-clear queuecounters")
+            global output_table
+
+            if queue_val is None:
+                logger.info("No queue found for dscp {} on DUT".format(inner_dscp))
+                output_table.append([inner_dscp, "No queue found", "N/A"])
+                continue
+
+            try:
+                send_and_verify_traffic(duthost=duthost,
+                                        ptfadapter=ptfadapter,
+                                        pkt=pkt,
+                                        exp_pkt=exp_pkt,
+                                        ptf_src_port_id=outer_src_port_id,
+                                        ptf_dst_port_id=inner_dst_port_id,
+                                        dut_egress_port=dut_egress_port,
+                                        egress_queue=queue_val,
+                                        inner_dscp=inner_dscp,
+                                        asic_type=dutTestParams["asic_type"])
+
+            except Exception as e:
+                raise (e)
+
+        logger.info("DSCP to queue mapping test results:\n{}"
```

Comment:
Can you please paste the output to the PR? Thanks!

Reply:
Pasted in the PR description